test/src/test/java/org/corfudb/integration/LogReplicationRoutingQueueIT.java

```diff
@@ -220,6 +223,54 @@ public void testRoutingQueue2way() throws Exception {
         });
     }
 
+    @Test
+    public void testSnapshotCancel() throws Exception {
+
+        // Register client and setup initial group destinations mapping
+        CorfuRuntime clientRuntime = getClientRuntime();
+        CorfuStore clientCorfuStore = new CorfuStore(clientRuntime);
+        String sourceSiteId = DefaultClusterConfig.getSourceClusterIds().get(0);
+        CorfuStoreBrowserEditor editor = new CorfuStoreBrowserEditor(clientRuntime, null, true);
+        String clientName = DEFAULT_ROUTING_QUEUE_CLIENT;
+        RoutingQueueSenderClient queueSenderClient = new RoutingQueueSenderClient(clientCorfuStore, clientName);
+        CountDownLatch startedTransmitting = new CountDownLatch(1);
+        SnapshotProvider snapshotProvider = new SnapshotProvider(clientCorfuStore, startedTransmitting);
+        queueSenderClient.startLRSnapshotTransmitter(snapshotProvider);
+
+        startReplicationServers();
+
+        RoutingQueueListener listener = new RoutingQueueListener(sinkCorfuStores.get(0),
+                sourceSiteId, clientName);
+        sinkCorfuStores.get(0).subscribeRoutingQListener(listener);
```

Comment:
subscription should be done before the replication servers are started to ensure that no replicated data update is missed.